### ISSUE TITLE
Creating a 'head' branch causes detachment

### ISSUE THREAD

**Issue by haykam821** [desktop/desktop#13638]
### Describe the bug

After creating a 'head' branch, GitHub Desktop changed to a detached HEAD mode, which can't be published.

### Version & OS

- GitHub Desktop 2.9.6
- macOS 12.1

### Steps to reproduce the behavior

1. Open the branch picker
2. Opt to create a new branch
3. Name the branch `head`
4. Create the branch
5. Observe the branch status

### Expected behavior

I expected to be able to use the new branch like a typical branch, or get an error when creating the branch.

### Actual behavior

The branch was not a normal branch but rather a detached HEAD.

### Screenshots

![](https://user-images.githubusercontent.com/24855774/148469778-cb961e02-0840-4c54-ac9a-21654f949bf6.png)

**Comment by sergiou87** [comment]
I don't think you should use `head` as a branch name, git doesn't seem to like it either since it's printing this all the time:
`warning: refname 'head' is ambiguous.`

We should probably not allow users to enter `head` as branch name 🤔

**Comment by Emadeazolli1989** [comment]
مرحبا جميعا آمل أن تكونوا بخير وعافيه واعلموا أن الشافي هو الله
اللهم صل وسلم على نبينا محمد وعلى آله وصحبه أجمعين
دمتم بخير
